src/reconcile/rec_visibility.c

```diff
@@ -475,9 +473,26 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
             /* The beginning of the validity window is the selected update's time point. */
             WT_TIME_WINDOW_SET_START(select_tw, upd);
         else if (select_tw->stop_ts != WT_TS_NONE || select_tw->stop_txn != WT_TXN_NONE) {
-            /* If we only have a tombstone in the update list, we must have an ondisk value. */
-            WT_ASSERT(session,
-              vpack != NULL && tombstone != NULL && !vpack->tw.prepare && last_upd->next == NULL);
+            /*
+             * We only have a tombstone on the update list or all the updates are from the same
+             * transaction.
+             */
+            WT_ASSERT(session, tombstone != NULL);
+
+            /* We must have an ondisk value and it can't be a prepared update. */
+            WT_ASSERT(session, vpack != NULL && vpack->type != WT_CELL_DEL && !vpack->tw.prepare);
+
+            /* Move the pointer to the last update on the update chain. */
+            for (last_upd = tombstone; last_upd->next != NULL; last_upd = last_upd->next)
```

Comment:
@agorrod I moved the initialization of last_upd here. It is slightly less efficient but it should make coverity happy. I don't expect we need to walk a lot of updates so I think this should be OK.